### PR TITLE
Convert datepicker to shadcn/ui component

### PR DESCRIPTION
**Pull request by NechamaSopher**

### CODE REVIEW

**Review by bambooch — CHANGES_REQUESTED** [review]
Hi @NechamaSopher thanks for the PR!

I'm getting the following error upon compiling, I've done `pnpm install` from the root project
image

perhaps something is missing? let me know if you have any idea

**Comment by bambooch** [comment]
also checked by running `pnpm install` and `npm install` from frontend with no luck, perhaps there are some dependencies missing? is it running on your side?

**Comment by NechamaSopher** [comment]
@bambooch By me it is all runing, I will try re installing and let you know.
from the error it looks like react-day-picker is missing try runing `npm install react-day-picker date-fns`.

**Comment by bambooch** [comment]
Hi @NechamaSopher now it works almost 100% correct. I noticed a bug on inital load just after running `pnpm turbo dev` from the root dir or `npm run dev` from the fronted project. 

You can see the result and error message in this jam here: https://jam.dev/c/a1734f18-e74c-413c-9b62-12b107622c4b

**Comment by NechamaSopher** [comment]
@bambooch So sorry about this. I do not get any errors.
Do you mean "Uncaught SyntaxError: Invalid or unexpected token"?

**Comment by bambooch** [comment]
> @bambooch So sorry about this. I do not get any errors. Do you mean "Uncaught SyntaxError: Invalid or unexpected token"?

Hey @NechamaSopher yes that is the one, but it's not just the console error the control doesn't work upon that initial load. 

Does it behave like this for you? Or not?

**Comment by bambooch** [comment]
Interesting that it is the same thing happening as on #109. I'll try deleting my local node_modules and reinstall everything to see if perhaps this has something to do with it...

**Comment by bambooch** [comment]
Ahh just wanted to commit this fix @NechamaSopher but cannot since you opened a PR from your forks main branch. For future reference always open PRs from a new branch on your fork so others can contribute as well. I'll try implementing it on #109 now and merge that first so maybe you can pull in those changes or open a new PR.

**Comment by NechamaSopher** [comment]
@bambooch Hi, So sorry I missed this, it is one of my first time contributing so did not know that. Thank you.
Sould I commit this change?

File: frontend/package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "@nextui-org/react": "^2.3.6",
     "@radix-ui/react-dialog": "^1.1.2",
     "@radix-ui/react-icons": "^1.3.0",
+    "@radix-ui/react-popover": "^1.1.2",
```
Comment (bambooch) [inline frontend/package.json:21]: there is exactly the same package two lines below

**Review by bambooch — CHANGES_REQUESTED** [review]

**Comment by bambooch** [comment]
@NechamaSopher the bug I encountered is a bit difficult to reproduce but it is definitely occuring due to the mix of older styling tech. your PR works but it just has a duplicate package as you can see in package.json, please remove it (since I can't push anything to your main branch, make sure to alawys make a new branch for future PRs)

**Comment by bambooch** [comment]
@NechamaSopher Since it's a bit cleaner and it allowed me to fix this bug I encountered, I merged @nine-codes version of the datepicker as part of #122. 

Since you're already familiar with the datepicker code, can you merge the progress that's been made on our main branch now to this PR and add a few quality changes to the datepicker (maybe make the popover bigger) or anything else really that improves so I can approve your PR and for it to have some valid contribution since you did take time to create a solid PR.

File: frontend/package.json (hunk 1 above)
Comment (NechamaSopher) [reply frontend/package.json:21]: In my local it fixed , I just didn't commit it.

**Comment by NechamaSopher** [comment]
@bambooch Hi, I'm trying to understand, this pr is not need any more because your project has already the datepicker as shadcn/ui  correct?

**Comment by bambooch** [comment]
> @bambooch Hi, I'm trying to understand, this pr is not need any more because your project has already the datepicker as shadcn/ui correct?

Hi @NechamaSopher yes but you can merge the progress made so far to this PR and add some customization to it as you see fit so I can make you a contributor since you did create something valid but edge case bugs prevented it from being fully merged.

Perhaps you can make the datepicker popover appear larger on mobile screens since it's a bit too small on mobile for my taste. Let me know if you need further guidance. You can also reach me probably faster on our Discord server (link in README)

**Pull request closed without merging**